### PR TITLE
[en]: description of the cronjob schedule timezone

### PR DESCRIPTION
**Pull request by cndoit18**
Signed-off-by: cndoit18 <cndoit18@outlook.com>

description of the cronjob schedule timezone

fixes: #29463

link: kubernetes/kubernetes#104404

### CODE REVIEW

File: content/en/docs/concepts/workloads/controllers/cron-jobs.md [hunk 1]
```diff
@@ -24,6 +24,7 @@ All **CronJob** `schedule:` times are based on the timezone of the
 If your control plane runs the kube-controller-manager in Pods or bare
 containers, the timezone set for the kube-controller-manager container determines the timezone
 that the cron job controller uses.
+To specify the CronJob schedule timezone, you can also use [TZ](https://en.wikipedia.org/wiki/List_of_tz_database_time_zones)
```
Comment (HenryKamg) [inline content/en/docs/concepts/workloads/controllers/cron-jobs.md:27]: To specify the timezone of CronJob schedule, you can get the list of timezone from [here](https://en.wikipedia.org/wiki/List_of_tz_database_time_zones),  and modify timezone name, such as `UTC`, in CronJob schedule as you need.

File: content/en/docs/concepts/workloads/controllers/cron-jobs.md [hunk 2]
```diff
@@ -53,15 +55,16 @@ takes you through this example in more detail).
...
+#   ┌──────────────── timezone (optional)
+#   |    ┌───────────── minute (0 - 59)
+#   |    │ ┌───────────── hour (0 - 23)
+#   |    │ │ ┌───────────── day of the month (1 - 31)
+#   |    │ │ │ ┌───────────── month (1 - 12)
+#   |    │ │ │ │ ┌───────────── day of the week (0 - 6) (Sunday to Saturday;
+#   |    │ │ │ │ │                                   7 is also Sunday on some systems)
+#   |    │ │ │ │ │
+#   |    │ │ │ │ │
+# TZ=UTC * * * * *
```
Comment (sftim) [inline content/en/docs/concepts/workloads/controllers/cron-jobs.md:67]: It looks like we're now using the format from https://pkg.go.dev/github.com/robfig/cron/v3#hdr-CRON_Expression_Format :sob:

The documented timezone variable there is `CRON_TZ`; `TZ` is listed as a legacy variable.

File: content/en/docs/concepts/workloads/controllers/cron-jobs.md [hunk 3]
```diff
@@ -24,6 +24,8 @@ All **CronJob** `schedule:` times are based on the timezone of the
 If your control plane runs the kube-controller-manager in Pods or bare
 containers, the timezone set for the kube-controller-manager container determines the timezone
 that the cron job controller uses.
+
+To specify the timezone of CronJob schedule, you can get the list of timezone from [here](https://en.wikipedia.org/wiki/List_of_tz_database_time_zones),  and modify timezone name, such as `UTC`, in CronJob schedule as you need.
```
Comment (sftim) [inline content/en/docs/concepts/workloads/controllers/cron-jobs.md:28]: I'd replace this caution callout with a note, just below the format description, that recommends **always** setting `CRON_TZ`. How does that sound?

I'm also tempted to replace the link to https://en.wikipedia.org/wiki/Cron with link to a new page in the reference section, documenting the Golang / Kubernetes flavor of that format. We can explain that the Golang flavor is derived from Vixie cron.

**Review by sftim — CHANGES_REQUESTED** [review]
Thanks for pointing this out @cndoit18 

It looks like we missed out updating this documentation for the v1.22 release.
/sig apps

File: content/en/docs/concepts/workloads/controllers/cron-jobs.md (hunk 2 above)
Comment (cndoit18) [reply content/en/docs/concepts/workloads/controllers/cron-jobs.md:67]: Thank you very much, I'll update it

File: content/en/docs/concepts/workloads/controllers/cron-jobs.md (hunk 3 above)
Comment (cndoit18) [reply content/en/docs/concepts/workloads/controllers/cron-jobs.md:28]: Thank you so much! It looks great!

**Comment by cndoit18** [comment]
hi, @sftim This is my first involvement with Kubernetes/website and I don't know much about the process involved.

Do I need to update the Kubernetes version above?
 As well as my broken English may not support a large English description. 😢 

I hope you can help me, thanks!

**Comment by jimangel** [comment]
No problem @cndoit18. Can you squash your commits please? This looks ready to merge otherwise. I wouldn't mind seeing a tech review / validation from someone.

/lgtm

**Comment by cndoit18** [comment]
> No problem @cndoit18. Can you squash your commits please? This looks ready to merge otherwise. I wouldn't mind seeing a tech review / validation from someone.
> 
> /lgtm

Done, thank you.

**Comment by soltysh** [comment]
This PR is wrong, we never state that this functionality is officially supported. I'll open a revert when I figure out the entire situation with this.

**Pull request merged**